### PR TITLE
Upgrade medusa-operator

### LINKED ISSUE #229: Upgrade medusa-operator and restore template
This is a reminder for follow up changes that will be needed after https://github.com/k8ssandra/k8ssandra/issues/192 is merged. Two things are needed:

* Update medusa-operator image
* Update the `restore/templates/restore.yaml` template

### PR DESCRIPTION
**Pull request by jsanda**
Fixes #229 and #226

This PR has the following changes:

* Updates medusa-operator image
* Update CassandraBackup CRD
* Update CassandraRestore CRD
* Updates restore template with latest changes in medusa-operator

### CODE REVIEW

File: charts/k8ssandra/crds/medusa/backup.yaml [hunk 1]
```diff
@@ -3817,14 +3722,7 @@ spec:
...
                       description: Version string for config builder, used to generate Cassandra server configuration
-                      enum:
-                        - 6.8.0
-                        - 6.8.1
-                        - 6.8.2
-                        - 6.8.3
-                        - 3.11.6
-                        - 3.11.7
-                        - 4.0.0
+                      pattern: (6\.8\.\d+)|(3\.11\.\d+)|(4\.0\.\d+)
```
Comment (emerkle826) [inline charts/k8ssandra/crds/medusa/backup.yaml:3725]: This is more of a question, but will `4.0-beta4` match `4.0`? The pattern will only handle digits it seems, so I guess I'm not sure where the version here maps to an image tag.

**Review by emerkle826 — APPROVED** [review]
This looks ok to me.

File: charts/k8ssandra/crds/medusa/backup.yaml (hunk 1 above)
Comment (jsanda) [reply charts/k8ssandra/crds/medusa/backup.yaml:3725]: That is a great question. It looks like the answer is no. We will have to open an issue in cass-operator. I can do that.

Comment (emerkle826) [reply charts/k8ssandra/crds/medusa/backup.yaml:3725]: will do

Comment (emerkle826) [reply charts/k8ssandra/crds/medusa/backup.yaml:3725]: done: https://github.com/datastax/cass-operator/issues/373

**Pull request merged**